config/initializers/secure_headers.rb

```diff
@@ -1,11 +1,27 @@
 # rubocop:disable Lint/PercentStringArray
 Rails.application.config.after_initialize do
   SecureHeaders::Configuration.default do |config|
-    config.cookies = {
+    cookies_config = {
       secure: true,
       httponly: true
     }
 
+    # SameSite cannot be set to None in test environment, because
+    # in test environment the app is not under HTTPS. Therefore,
+    # Set-Cookie header with session id will be discarded by the browser
+    # and feature specs will fail.
+    # But having it set for other envs means that other envs have to be
+    # used under HTTPS.
+    # So, probably it should be configured by a specific setting or
+    # dynamically (for instance when there are some active nextcloud storages
+    # enabled)
+    unless Rails.env.test?
+      cookies_config[:samesite] = {
+        none: true
```

Comment:
@ba1ash did you and @wielinde / the team already discuss potential solutions for this? Setting samesite=none as a default ist not an option that we can ship. It must be opt-in, or secured by some other authentication measure. I'd strongly vote for using a separate authentication mechanisms for iframes, to ensure the session cookie is always samesite.

Reply:
The worst part about it: It does not solve the problem apparently to share the session cookie with the iframe. Yesterday's conclusion is that the iframe integration will not work as we thought as the session cannot be shared.

Reply:
@oliverguenther you're absolutely right. We discussed it. The PR is a PoC. And like @wielinde stated it does not satisfy requirements we have.